File: Axiom/Internal/Intermediate/ImExpressionStatement.cs
using System.IO;

namespace Axiom.Internal.Intermediate
{
    internal class ImExpressionStatement : ImStatement
    {
        public ImExpressionStatement(ImExpression expression)
        {
            Expression = expression;
        }

        public override void Dump(TextWriter tw, int ident)
        {
            Report.Dump(tw, ident, "ExpressionStatement");
            Expression.Dump(tw, ident + 1);
        }

        public override ImSequence Linear()
        {
            var linear = new ImSequence();
            var linearExpr = Expression.Linear();

            linear.Statements.AddAll(((ImSequence)linearExpr.Statement).Statements);
            linear.Statements.Add(new ImExpressionStatement(linearExpr.Expression));

            return linear;
        }

        public ImExpression Expression { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/im/ImExpressionStatement.java
package com.moybl.axiom.im;

public class ImExpressionStatement implements ImStatement {

	private ImExpression expression;

	public ImExpressionStatement(ImExpression expression) {
		this.expression = expression;
	}

	@Override
	public void accept(ImVisitor visitor) {
		visitor.visit(this);
	}

	public ImExpression getExpression() {
		return expression;
	}

}
